app/vminsert/promremotewrite/request_handler.go

```diff
@@ -71,6 +71,20 @@ func insertRows(at *auth.Token, timeseries []prompb.TimeSeries, extraLabels []pr
 		}
 		perTenantRows[*atLocal] += len(ts.Samples)
 	}
+
+	var atLocal *auth.Token
+	for i := range mms {
+		m := &mms[i]
+		if atLocal == nil || m.AccountID != atLocal.AccountID || m.ProjectID != atLocal.ProjectID {
```

Comment:
```suggestion
	for i := range mms {
		m := &mms[i]
		atLocal := at
		if atLocal == nil {
```
The `at` should have higher priority than metadata row tenant info if both exist.